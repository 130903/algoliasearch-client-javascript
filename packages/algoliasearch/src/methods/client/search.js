// @flow

type SearchRequest = {| indexName: IndexName, params: Parameters |};

export default function search(req: RequestMethod, requests: SearchRequest[]) {
  return req({
    method: 'POST',
    path: '/1/indexes/*/queries',
    body: { requests },
  });
}
