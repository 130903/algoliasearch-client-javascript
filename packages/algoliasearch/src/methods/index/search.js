// @flow
import type { RequestMethod, IndexName, Parameters } from '../../../types';

export default function search(
  req: RequestMethod,
  indexName: IndexName,
  params: Parameters
) {
  return req({
    method: 'POST',
    path: `/1/indexes/${indexName}/query`,
    body: { params },
  });
}
