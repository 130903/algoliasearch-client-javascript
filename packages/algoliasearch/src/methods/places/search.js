// @flow
import type { RequestMethod, Parameters } from '../../types';

export default function search(req: RequestMethod, params: Parameters) {
  return req({
    method: 'POST',
    path: `/1/indexes/places/query`,
    body: { params },
  });
}
