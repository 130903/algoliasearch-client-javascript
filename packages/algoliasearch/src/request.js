// @flow
import type { Requester, AppId, ApiKey } from './types';

// it will simply give back the arguments given
// this is a mocked method, not implemented yet
// later, we'll import from 'algoliasearch-request'
//
/* eslint-disable no-unused-vars */
export const createRequester: Requester = (
  appId: AppId,
  apiKey: ApiKey
) => requestParams => new Promise(resolve => resolve(requestParams));
