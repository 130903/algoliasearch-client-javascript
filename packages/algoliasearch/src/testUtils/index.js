// @flow

import type { RequestMethod } from '../types';

export const snapshotAll = requests =>
  requests.map(req => req.then(sn => expect(sn).toMatchSnapshot()));

export const fakeRequester: RequestMethod = requestParams =>
  new Promise(resolve => resolve(requestParams));
