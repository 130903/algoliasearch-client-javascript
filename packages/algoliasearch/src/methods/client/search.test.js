import { snapshotAll } from '../../testUtils';
import search from './search';

import { createRequester } from '../../request';
const req = createRequester();

it('search', () => {
  const requests = [
    search(req, [
      {
        indexName: 'some_index',
        params: {},
      },
    ]),
  ];

  snapshotAll(requests);
});
