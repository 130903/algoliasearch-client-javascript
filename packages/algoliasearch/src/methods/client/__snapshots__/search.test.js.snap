// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search 1`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "indexName": "some_index",
        "params": Object {},
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes/*/queries",
}
`;
